components/mailchimp/sources/new-segment-tag-subscriber/new-segment-tag-subscriber.js

```diff
@@ -0,0 +1,97 @@
+const common = require("../common-webhook");
+const { mailchimp } = common.props;
+
+module.exports = {
+  ...common,
+  key: "mailchimp-new-segment-tag-subscriber",
+  name: "New Segment Tag Subscriber",
+  description:
+    "Emit an event when an subscriber is added to a segment or tags within an audience list.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    mailchimp,
+    server: { propDefinition: [mailchimp, "server"] },
+    listId: {
+      type: "string",
+      label: "Audience List Id",
+      description:
+        "The unique ID of the audience list related to the segment or tag you'd like to watch for new subscribers.",
+    },
+    segmentId: {
+      type: "string",
+      label: "Segment/Tag Id",
+      description:
+        "The unique ID of the segment or tag you'd like to watch for new subscribers.",
+    },
+    includeTransactional: {
+      type: "boolean",
+      label: "Include subscribers from Mailchimp Transactional?",
+      description:
+        "If set to `true`, it will include subscribers from Mailchimp Marketing and Mailchimp Transactional, formerly Mandrill.  When set to `false`, it will include subscribers from Mailchimp Marketing only.",
+      default: false,
+      optional: true,
+    },
+    timer: { propDefinition: [mailchimp, "timer"] },
```

Comment:
The `common-webhook` module is supposed to contain reusable code for http-based event sources. However, it is being used by timer-based event sources. This is counterintuitive and will hurt maintainability. I suggest we refactor the reusable code in a similar way to [the Sendgrid event sources](https://github.com/PipedreamHQ/pipedream/tree/master/components/sendgrid/sources/common)

Reply:
This has been previously been addressed.